### DISCUSSION TITLE
gatsby-source-graphql doesn't work with Gatsby Preview

### DISCUSSION THREAD

**Discussion by MotaZor94** [gatsbyjs/gatsby#32419] [Help]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

Hi everyone, it seems that when I use Strapi with Gatsby-source-graphql the data doesn't change in Gatsby Preview even tho the rebuild happened on the Gatsby Preview server.

### Reproduction Link

N/A

### Steps to Reproduce

1.Create gatsby project
2. Install gatsby-source-graphql
3. Setup Gatsby Preview on gatsby cloud
...

### Expected Result

I'm expecting the data to change once I trigger the webhook

### Actual Result

No data has changed but the preview has been rebuilt

### Environment

```shell
System:
    OS: macOS 11.2.3
    CPU: (4) x64 Intel(R) Core(TM) i5-7360U CPU @ 2.30GHz
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.16.1 - /usr/local/bin/node
    Yarn: 1.19.2 - /usr/local/bin/yarn
    npm: 6.14.12 - /usr/local/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 91.0.4472.114
    Safari: 14.0.3
  npmPackages:
    gatsby: 3.6.2 => 3.6.2
    gatsby-plugin-eslint: 3.0.0 => 3.0.0
    gatsby-plugin-image: ^1.1.0 => 1.1.0
    gatsby-plugin-manifest: 3.0.0 => 3.0.0
    gatsby-plugin-offline: 4.0.0 => 4.0.0
    gatsby-plugin-postcss: ^4.0.0 => 4.0.0
    gatsby-plugin-react-helmet: 4.0.0 => 4.0.0
    gatsby-plugin-remove-fingerprints: 0.0.2 => 0.0.2
    gatsby-plugin-sharp: ^3.0.1 => 3.0.1
    gatsby-plugin-styled-components: ^4.6.0 => 4.6.0
    gatsby-source-filesystem: ^3.1.0 => 3.1.0
    gatsby-source-graphql: ^3.6.0 => 3.6.0
    gatsby-source-strapi: ^1.0.2 => 1.0.2
    gatsby-transformer-sharp: ^3.1.0 => 3.1.0
  npmGlobalPackages:
    gatsby-cli: 3.6.0
```

### Config Flags

PRESERVE_FILE_DOWNLOAD_CACHE: true,
PRESERVE_WEBPACK_CACHE: true

**Comment by LekoArts** [comment]
Hi! This is working as expected as it doesn't support it.

`gatsby-source-graphql` skips the Gatsby Node model altogether, and allows you to directly pull data from your GraphQL API. At first this seems really attractive as things just work, but as your site grows beyond a handful of pages, a few problems arise.

- There's no way to use Gatsby Preview effectively as the Preview instance will have to run every query in your Gatsby site again. This means in many cases you would need to wait nearly the length of an entire uncached build for your preview to show up.
- There's no way to cache data. This means incremental builds can't work their magic, but it also means regular builds are very slow. Every content change in your API requires all the data to be refetched, making things very slow.

**Reply by MotaZor94** [reply]
Hi Leko,

Thanks for your response. So what would be the other alternative to create a preview functionality? We already created a full system and redoing everything with for example gatsby-source-strapi would be a lot of work to do - just to get a preview functionality.
